### PR TITLE
Feature: bypass install screen and allow custom SECRET_KEY definition related #13

### PR DESCRIPTION
**Pull request by DrMegavolt**
@jfelten this is part one: to define custom secret key and skip install step

### CODE REVIEW

File: templates/gitea/gitea-config.yaml [hunk 1]
```diff
@@ -280,9 +280,9 @@ data:
 
     [security]
     ; Whether the installer is disabled
-    INSTALL_LOCK = false
+    INSTALL_LOCK = {{ .Values.config.disableInstaller }}
     ; !!CHANGE THIS TO KEEP YOUR USER DATA SAFE!!
-    SECRET_KEY = !#@FDEWREWR&*(
+    SECRET_KEY = {{ .Values.config.secretKey }}
```
Comment (jfelten) [inline templates/gitea/gitea-config.yaml:286]: Can this be set to a random password if not specified? 
Like this:
{{ if .Values.config.secretKey }}
SECRET_KEY = {{ .Values.config.secretKey }}
{{ else }}
SECRET_KEY = {{ randAlphaNum 10 | b64enc | quote }}
{{ end }}

File: values.yaml [hunk 2]
```diff
@@ -140,3 +140,8 @@ affinity: {}
 ## Annotations for the deployment and nodes.
 deploymentAnnotations: {}
 podAnnotations: {}
+
+## In order to disable initial install screen you must have secretKey and disableInstaller=true
+config:
+  secretKey: defaultsecretkeyyouneedtooverride
```
Comment (jfelten) [inline values.yaml]: Comment this out to leave it unset by default
#secretKey: defaultsecretkeyyouneedtooverride

**Comment by jfelten** [comment]
Please add documentation in README.md for the new values like this:

| `config.secretKey`                    | `gitea` config secretKey                    | `set to a random password`                                                  |

**Review by jfelten — CHANGES_REQUESTED** [review]
Please see inline comments - when those are addressed LGTM

File: templates/gitea/gitea-config.yaml (hunk 1 above)
Comment (DrMegavolt) [reply templates/gitea/gitea-config.yaml:286]: I've checked Gitea helper CLI commands. They use 64 random string. 
https://github.com/go-gitea/gitea/blob/master/modules/generate/generate.go#L74 
I did the same

**Comment by DrMegavolt** [comment]
@jfelten  doc updated; I did the key generation similar to Gitea CLI helper for consistency

**Review by jfelten — APPROVED** [review]
LGTM

**Pull request merged**